Maintain a cache of repo remotes that we know about

alexr00 opened the pull request:
Part of #271101

alexr00 committed:
Maintain a cache of repo remotes that we know about
Part of #271101

alexr00 committed:
Save workspace file or folder instead of repo root path

lszomoru commented on extensions/git/src/knownFolders.ts:
```diff
@@ -0,0 +1,143 @@
+	}
+
+	/**
+	 * We should possibly support converting between ssh remotes and http remotes.
```
I would probably "normalize" the remote when storing it but we can tackle this later.

lszomoru commented on extensions/git/src/knownFolders.ts:
```diff
@@ -0,0 +1,143 @@
+	private static readonly EXPIRY_MS = 90 * 24 * 60 * 60 * 1000; // 90 days in ms
+
+	// Outer LRU: repoUrl -> inner LRU (folderPathOrWorkspaceFile -> timestamp ms). Only keys matter externally, timestamp tracks last seen.
+	private readonly lru = new LRUCache<string, LRUCache<string, number>>(KnownFolders.MAX_REPO_ENTRIES);
```
Should the "inner" cache use URI instead of a string?

alexr00 commented on extensions/git/src/knownFolders.ts:
```diff
@@ -0,0 +1,143 @@
+	private static readonly STORAGE_KEY = 'git.knownFolders';
+	private static readonly MAX_REPO_ENTRIES = 30; // Max repositories tracked
+	private static readonly MAX_FOLDER_ENTRIES = 10; // Max folders per repository
+	private static readonly EXPIRY_MS = 90 * 24 * 60 * 60 * 1000; // 90 days in ms
```
It seemed nice to have, but no, we don't need it.

alexr00 committed:
PR feedback
- Rename to repositoryCache
- Remove timestamp
- Move logic into repo cache
- Fix tests

alexr00 committed:
Name changes

alexr00 committed:
Actually delete

alexr00 committed:
Fix change that got lost in the rename
